azure/functions/decorators/function_app.py

```diff
@@ -3297,8 +3287,27 @@ def get_functions(self) -> List[Function]:
                 '-bindings-http-webhook-trigger?tabs=in-process'
                 '%2Cfunctionsv2&pivots=programming-language-python#http-auth')
 
+        FunctionRegister.validate_functions(self, functions=functions)
+
         return functions
 
+    def validate_functions(self, functions: List[Function]) -> bool:
```

Comment:
We can mark this as static

Reply:
I don't think it can be static.

The `@staticmethod` decorator removes the `self` param, which is needed for accessing the function names dict.
I also feel it shouldn't be moved out of the `FunctionRegister` class, since it fits within the logic and the function names dict belongs to that class.